store/tikv/scan.go

```diff
@@ -98,7 +100,8 @@ func (s *Scanner) Next() error {
 			s.Close()
 			return errors.Trace(err)
 		}
-		if len(s.Value()) == 0 {
+
+		if len(s.Value()) == 0 && !s.keyOnly {
```

Comment:
As I said `resolveCurrentLock` uses `snapshotGet` to resolve lock. When `len(s.Value()) == 0`, it means the key does not exist. It has nothing to do with the `keyOnly` flag.